src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractGenerateNuGetPackageCommand.cs

```diff
@@ -23,18 +40,66 @@ protected AbstractGenerateNuGetPackageCommand(UnconfiguredProject unconfiguredPr
 
         protected override Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree node, bool focused, string commandText, CommandStatus progressiveStatus) =>
             ShouldHandle(node) ?
-                GetCommandStatusResult.Handled(GetCommandText(), CommandStatus.Enabled) :
+                GetCommandStatusResult.Handled(GetCommandText(), IsReadyToBuild() ? CommandStatus.Enabled : CommandStatus.Supported) :
                 GetCommandStatusResult.Unhandled;
 
+        private bool IsReadyToBuild()
+        {
+            return ErrorHandler.Succeeded(_buildManager.QueryBuildManagerBusy(out int busy)) && busy == 0;
+        }
+
         protected override async Task<bool> TryHandleCommandAsync(IProjectTree node, bool focused, long commandExecuteOptions, IntPtr variantArgIn, IntPtr variantArgOut)
         {
-            if (!ShouldHandle(node)) return false;
+            if (!ShouldHandle(node))
```

Comment:
`StartSimpleUpdateProjectConfiguration` bails out if anything is currently building, but it doesn't hurt to check IsReadyToBuild upfront in the handler. I'll add that check.